EventStore/EventStore-Client-Dotnet
# DEV-303 - Client certificate authentication for users

Sample code:

```
    class Program
    {
        static async Task Main(string[] args)
        {
            var certPath = "/path/to/user.crt";
            var certKeyPath = "/path/to/user.key";
            var connectionString = $"esdb://127.0.0.1:2113,127.0.0.2:2113,127.0.0.3:2113?tls=true&tlsVerifyCert=false&certPath={certPath}&certKeyPath={certKeyPath}";
            
            var settings = EventStoreClientSettings.Create(connectionString);
            
            using (var client = new EventStoreClient(settings)) {
                await client.SubscribeToStreamAsync(
                    streamName: "test",
                    start: FromStream.Start,
                    eventAppeared: (sub, evt, ct) => {
                        Console.WriteLine(evt.Event.EventStreamId + " " + evt.OriginalEventNumber);
                        return Task.CompletedTask;
                    });

                Console.ReadLine();
            }
        }
    }
```

Manual tests carried out:
- tested with disabled/deleted user
- tested multiple requests from same HTTP/2 connection (to see if caching is working properly)
- tested intermediate certificates (supported by adding intermediate CAs to the store)
- tested HTTP/1 fallback connection (with persistent subscription client)
- tested expired certificate


TODO:
- Add tests
- Automatic certificate rotation by detecting file changes (also requires refreshing of the HTTP connection pool)
